src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -341,42 +341,64 @@ public HttpClientSettingsKey getHttpClientKey() throws SnowflakeSQLException {
-      // Java implementation will work.
-      if (httpUseProxy
-          && !Strings.isNullOrEmpty(httpsProxyHost)
-          && !Strings.isNullOrEmpty(httpsProxyPort)) {
-        int proxyPort;
-        try {
-          proxyPort = Integer.parseInt(httpsProxyPort);
-        } catch (NumberFormatException | NullPointerException e) {
-          throw new SnowflakeSQLException(
-              ErrorCode.INVALID_PROXY_PROPERTIES, "Could not parse port number");
+        // There are 2 possible parameters for non proxy hosts that can be combined into 1
+        String combinedNonProxyHosts = Strings.isNullOrEmpty(nonProxyHosts) ? "" : nonProxyHosts;
+        if (!Strings.isNullOrEmpty(noProxy)) {
+          if (!Strings.isNullOrEmpty(combinedNonProxyHosts)) {
+            combinedNonProxyHosts += "|";
+          }
+          combinedNonProxyHosts += noProxy;
+        }
+        if (!Strings.isNullOrEmpty(httpsProxyHost) && !Strings.isNullOrEmpty(httpsProxyPort)) {
+          int proxyPort;
+          try {
+            proxyPort = Integer.parseInt(httpsProxyPort);
+          } catch (NumberFormatException | NullPointerException e) {
+            throw new SnowflakeSQLException(
+                ErrorCode.INVALID_PROXY_PROPERTIES, "Could not parse port number");
+          }
+          ocspAndProxyKey =
+              new HttpClientSettingsKey(
+                  getOCSPMode(),
+                  httpsProxyHost,
+                  proxyPort,
+                  combinedNonProxyHosts,
+                  "", /* user = empty */
+                  "", /* password = empty */
+                  "https");
+        } else if (!Strings.isNullOrEmpty(httpProxyHost) && !Strings.isNullOrEmpty(httpProxyPort)) {
+          int proxyPort;
+          try {
+            proxyPort = Integer.parseInt(httpProxyPort);
+          } catch (NumberFormatException | NullPointerException e) {
+            throw new SnowflakeSQLException(
+                ErrorCode.INVALID_PROXY_PROPERTIES, "Could not parse port number");
+          }
+          ocspAndProxyKey =
+              new HttpClientSettingsKey(
+                  getOCSPMode(),
+                  httpProxyHost,
+                  proxyPort,
+                  combinedNonProxyHosts,
+                  "", /* user = empty */
+                  "", /* password = empty */
+                  "http");
         }
-        ocspAndProxyKey =
-            new HttpClientSettingsKey(
-                getOCSPMode(),
-                httpsProxyHost,
-                proxyPort,
-                noProxy,
-                "", /* user = empty */
-                "", /* password = empty */
-                "https");
       } else {
         // If no proxy is used or JVM http proxy is used, no need for setting parameters
+        logger.debug("http.useProxy={}. JVM proxy not used.", httpUseProxy);
         ocspAndProxyKey = new HttpClientSettingsKey(getOCSPMode());
```

Comment:
What happens if httpUseProxy is true but all other settings are empty? Previously, this line will be executed but the new code the `ocspAndProxyKey` is not set. Am I right?

Reply:
This is true! Thanks for pointing out! I added an extra else clause to take care of that case.